fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/cascades/CostAndEnforcerJob.java

```diff
@@ -164,74 +167,100 @@ public void execute() {
+        }
+        StatsCalculator.estimate(groupExpression);
+        curTotalCost -= curNodeCost;
+        curNodeCost = CostCalculator.calculateCost(groupExpression);
+        groupExpression.setCost(curNodeCost);
+        curTotalCost += curNodeCost;
+
+        // record map { outputProperty -> outputProperty }, { ANY -> outputProperty },
+        recordPropertyAndCost(groupExpression, outputProperty, PhysicalProperties.ANY,
+                requestChildrenProperties);
+        recordPropertyAndCost(groupExpression, outputProperty, outputProperty, requestChildrenProperties);
+        enforce(outputProperty, requestChildrenProperties);
+        return true;
+    }
+
+    /**
+     * add enforce node
+     * @param outputProperty the group expression's out property
+     * @param requestChildrenProperty the group expression's request to its child.
+     */
     private void enforce(PhysicalProperties outputProperty, List<PhysicalProperties> requestChildrenProperty) {
         PhysicalProperties requiredProperties = context.getRequiredProperties();
-        if (!outputProperty.satisfy(requiredProperties)) {
-            EnforceMissingPropertiesHelper enforceMissingPropertiesHelper
-                    = new EnforceMissingPropertiesHelper(context, groupExpression, curTotalCost);
-            PhysicalProperties addEnforcedProperty = enforceMissingPropertiesHelper
-                    .enforceProperty(outputProperty, requiredProperties);
-            curTotalCost = enforceMissingPropertiesHelper.getCurTotalCost();
-
-            // enforcedProperty is superset of requiredProperty
-            if (!addEnforcedProperty.equals(requiredProperties)) {
-                recordPropertyAndCost(groupExpression.getOwnerGroup().getBestPlan(addEnforcedProperty),
-                        addEnforcedProperty, requiredProperties, Lists.newArrayList(outputProperty));
-            }
-        } else {
+        if (outputProperty.satisfy(requiredProperties)) {
             if (!outputProperty.equals(requiredProperties)) {
                 recordPropertyAndCost(groupExpression, outputProperty, requiredProperties, requestChildrenProperty);
             }
+            return;
+        }
+        EnforceMissingPropertiesHelper enforceMissingPropertiesHelper
+                = new EnforceMissingPropertiesHelper(context, groupExpression, curTotalCost);
+        PhysicalProperties addEnforcedProperty = enforceMissingPropertiesHelper
+                .enforceProperty(outputProperty, requiredProperties);
+        curTotalCost = enforceMissingPropertiesHelper.getCurTotalCost();
+
+        // enforcedProperty is superset of requiredProperty
+        if (!addEnforcedProperty.equals(requiredProperties)) {
+            recordPropertyAndCost(groupExpression.getOwnerGroup().getBestPlan(addEnforcedProperty),
+                    addEnforcedProperty, requiredProperties, Lists.newArrayList(outputProperty));
         }
     }
 
+    /**
+     * record property and cost
```

Comment:
should add a blank line between method doc and param docs. plz fix it in next PR